libsolidity/formal/SMTChecker.h

```diff
@@ -163,9 +163,44 @@ class SMTChecker: private ASTConstVisitor
 	VariableIndices visitBranch(ASTNode const* _statement, smt::Expression const* _condition = nullptr);
 	VariableIndices visitBranch(ASTNode const* _statement, smt::Expression _condition);
 
+	using CallStackEntry = std::pair<CallableDeclaration const*, ASTNode const*>;
+
+	/// Verification targets.
+	//@{
+	struct VerificationTarget
+	{
+		enum class Type { ConstantCondition, Underflow, Overflow, UnderOverflow, DivByZero, Balance, Assert } type;
+		smt::Expression value;
+		smt::Expression constraints;
+		Expression const* expression;
```

Comment:
What is the difference between value and expression?

Reply:
In many cases none, but for some checks the condition to be checked is different from the visited `solidity::Expression`.
Example:
Balance check during transfer function
```
        addVerificationTarget(
            VerificationTarget::Type::Balance,
            thisBalance < expr(*value),     
            &_funCall
        );
```

Div by zero
```
        addVerificationTarget(
            VerificationTarget::Type::DivByZero,
            _right,
            &_expression
        );
```

This is subtle and could be combined if required. Could maybe use only location instead of the Expression.

Reply:
Ah it's a solidity expression! Now I get it!